Packages/QRScanner/Sources/QRScannerView.swift

```diff
@@ -0,0 +1,135 @@
+// Copyright (c). Gem Wallet. All rights reserved.
+
+import SwiftUI
+import PhotosUI
+
+@MainActor
+public struct QRScannerView: View {
+    @Environment(\.openURL) private var openURL
+    @Environment(\.dismiss) private var dismiss
+
+    @State private var model: QRScanner
+
+    private let textProvider: QRScannerTextProviding
+    private let action: ((String) -> Void)
+
+    public init(action: @escaping ((String) -> Void)) {
+        self.init(action: action, textProvider: QRScannerTextProvider())
+    }
+
+    public init(action: @escaping ((String) -> Void), textProvider: QRScannerTextProviding) {
```

Comment:
`textProvider: QRScannerTextProviding` is overkill here, no need to add such complexity